### PR TITLE
Test for `email_notifier.py`

### LINKED ISSUE #47: Write tests for `email.py`
# Description
Develop a test file (`test_email.py`) to validate the functionality of the `email.py` script. The tests should ensure the script operates as expected under various scenarios, including normal conditions, edge cases, and error handling. The tests should use mock objects to simulate external dependencies e.g. the database and AWS SES, ensuring the tests remain isolated and reliable.

# Outcome
`test_email.py`: A Python test suite that uses `pytest` to validate the behaviour of the `email.py` script.

# User Story
As a developer, I want a robust test suite for `email.py` to ensure the script sends email alerts reliably and handles edge cases gracefully, so I can confidently deploy the script in production.

### PR DESCRIPTION
**Pull request by indiah444**
- Renamed `email.py` to `email_notifier.py` to fix error in standard import names.
- Started tests for `email_notifier.py`.

This (partly) closes #47

### CODE REVIEW

File: pipeline/test_email_notifier.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+from boto3.exceptions import Boto3Error
+from email_notifier import (
+    get_ses_client,
+    get_subscriptions_and_products,
+    send_email,
+    check_and_notify
+)
+
+
+def test_get_ses_client_missing_env_variables():
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:15]: You covered missing environment variables case which is good!

File: pipeline/test_email_notifier.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+        subscriptions = get_subscriptions_and_products(mock_connection)
+
+    assert len(subscriptions) == 1
+    assert subscriptions[0] == (1, 10.0, "Product A", "user@example.com")
+
+
+def test_send_email_success():
+    """Test sending an email using SES."""
+    mock_ses_client = MagicMock()
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:62]: used mocking effectively!

File: pipeline/test_email_notifier.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    with patch("email_notifier.get_cursor") as mock_get_cursor:
+        mock_get_cursor.return_value.__enter__.return_value = mock_cursor
+
+        subscriptions = get_subscriptions_and_products(mock_connection)
+
+    assert len(subscriptions) == 1
+    assert subscriptions[0] == (1, 10.0, "Product A", "user@example.com")
+
+
+def test_send_email_success():
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:60]: Good use of patching!

File: pipeline/test_email_notifier.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    """Test sending an email using SES."""
+    mock_ses_client = MagicMock()
+    mock_ses_client.send_email.return_value = {
+        "ResponseMetadata": {"HTTPStatusCode": 200}
+    }
+
+    with mock.patch.dict("os.environ", {
+        "FROM_EMAIL": "test@example.com",
+        "AWS_ACCESS_KEY_ID": "fake_key_id",
+        "AWS_ACCESS_SECRET_KEY": "fake_secret_key"
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:70]: Optional: Minor - I think this is repeated a few times, could this be refactored into a `@pytest.fixure`? - I know theres still a few things to sort with the tests so this can be considered at the end!

Something like:
```
@pytest.fixture
def aws_test_credentials():
    with mock.patch.dict("os.environ", {
        "FROM_EMAIL": "test@example.com",
        "AWS_ACCESS_KEY_ID": "fake_key_id",
        "AWS_ACCESS_SECRET_KEY": "fake_secret_key"})

Comment (gem09lo) [reply pipeline/test_email_notifier.py:70]: Im not 100% this code I wrote above works by the way! Just a rough example!

File: pipeline/test_email_notifier.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        Source="test@example.com",
+        Destination={"ToAddresses": ["user@example.com"]},
+        Message={
+            "Subject": {"Data": "Subject"},
+            "Body": {"Text": {"Data": "Body"}}
+        }
+    )
+
+
+def test_send_email_failure():
```
Comment (gem09lo) [inline pipeline/test_email_notifier.py:84]: Nice and descriptive function names!

File: pipeline/test_email_notifier.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+def test_get_ses_client_missing_env_variables():
+    """Test if an exception is raised when required environment variables are missing."""
+    with mock.patch.dict("os.environ", {}, clear=True):
+        with pytest.raises(RuntimeError, match="Required environment variables for AWS or email are missing"):
+            get_ses_client()
+
+
+def test_get_ses_client_success():
+    """Test the successful creation of an SES client."""
+    with mock.patch.dict("os.environ", {
```
Comment (anitamegarry) [inline pipeline/test_email_notifier.py:24]: Great that env variables are being mocked

**Review by anitamegarry — APPROVED** [review]
Looks good to me!!!
